apps/guardian-ui/src/components/meta/MetaManager.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useState } from 'react';
+import { Box, Button, Flex, Text } from '@chakra-ui/react';
+import { useTranslation, metaToHex, fieldsToMeta } from '@fedimint/utils';
+import { MetaConfig, MetaFields } from '@fedimint/types';
+import { ModuleRpc } from '../../types';
+import { useAdminContext } from '../../hooks';
+import { EditMetaField } from './EditMetaField';
+import { ReviewMetaField } from './ReviewMetaField';
+import { ViewConsensusMeta, ConsensusMetaFields } from './ViewConsensusMeta';
+
+const DEFAULT_META_KEY = 0;
+const POLL_TIMEOUT_MS = 2000;
+
+interface MetaManagerProps {
+  metaModuleId?: string;
+  metaConfig: MetaConfig;
+}
+
+export const MetaManager = React.memo(function MetaManager({
+  metaModuleId,
+}: MetaManagerProps): JSX.Element {
+  const { t } = useTranslation();
+  const { api } = useAdminContext();
+
+  const [consensusMeta, setConsensusMeta] = useState<ConsensusMetaFields>();
+  const [editedMetaFields, setEditedMetaFields] = useState<MetaFields>([]);
+
+  const editConsensusMeta = useCallback(() => {
+    if (consensusMeta) {
+      setEditedMetaFields(consensusMeta.value);
+    }
+  }, [consensusMeta]);
+
+  const proposeMetaEdits = useCallback(() => {
+    if (metaModuleId === undefined) {
+      return;
+    }
+
+    api
+      .moduleApiCall<{ metaValue: string }[]>(
+        Number(metaModuleId),
+        ModuleRpc.submitMeta,
+        {
+          key: DEFAULT_META_KEY,
+          value: metaToHex(fieldsToMeta(editedMetaFields)),
+        }
+      )
+      .then(() => setEditedMetaFields([]))
+      .catch(console.error);
+  }, [api, metaModuleId, editedMetaFields]);
+
+  return metaModuleId ? (
+    <Flex direction='column' gap={6}>
+      <ViewConsensusMeta
+        metaKey={DEFAULT_META_KEY}
+        metaModuleId={metaModuleId}
+        consensusMeta={consensusMeta}
+        updateConsensusMeta={(meta: ConsensusMetaFields) =>
+          setConsensusMeta(meta)
+        }
+        pollTimeout={POLL_TIMEOUT_MS}
+      />
+      <ReviewMetaField
+        metaModuleId={metaModuleId}
+        updateEditedMetaFields={(fields: MetaFields) => {
+          setEditedMetaFields([...editedMetaFields, ...fields]);
+        }}
+        pollTimeout={POLL_TIMEOUT_MS}
+      />
+      <Box>
+        <Text>
+          {t('federation-dashboard.config.manage-meta.edit-meta-label')}
+        </Text>
+        <EditMetaField
+          metaFields={editedMetaFields}
+          onChangeMetaFields={setEditedMetaFields}
+          protectDerivedMeta={false}
+        />
```

Comment:
I think it'd make sense to combine `ReviewMetaField` and `EditMetaField`. This could be done in a number of ways. For example:

1. Handle the data fetching/refetching in `MetaManager`.
2. Move all the form components into `ReviewMetaField` (Inputs and Buttons).
3. And pass the additional props to `ReviewMetaField` (from `MetaManager`): 
  * `metaFields` (fetched from the server) 
  * `isEditing` (to make the input boxes editable) 
  * `isRefetching` (to render a loading icon & disable further edits) 
  * `handleSubmitUpdates`(callback function to call onSubmit)
4. Handle the "draft" state within `ReviewMetaField` and pass the draft into `handleSubmitState` when the user clicks submit.

Happy to help implement this if you'd like :)

Reply:
Great suggestion. Go for it :)

Reply:
now making an attempt at this UX